src/containers/UserAddrWrapper.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react'
+import history from 'History'
+import { connect } from 'react-redux'
+import { withRouter } from 'react-router'
+import { geocode } from 'actions/Geocode'
+import setGeoQuery from 'actions/Geoquery.jsx'
+import UserAddr from 'components/UserAddr'
+import NavBarSearch from 'components/NavBarSearch'
+
+const mapStateToProps = state => {
+  return {
+    userCoords: [state.geocodeReducer.lat, state.geocodeReducer.lng],
+    address: state.geocodeReducer.address,
+    queryAddress: state.geoQueryReducer.address
+  }
+}
+
+const mapDispatchToProps = dispatch => {
+  return {
+    updateQueryInfo: type => dispatch(setGeoQuery(type)),
+    geocode: address => dispatch(geocode(address))
+  }
+}
+
+class UserAddrWrapper extends Component {
+  componentDidMount() {
+    // Clears GeoQuery to make top bar search disappear
+    if (this.props.version === 'Main') {
+      //this.props.updateQueryInfo('CLEAR_GEOQUERY')
+    }
+  }
+
+  updateQueryInfo = async (address, range) => {
+    await this.props.geocode(address)
```

Comment:
See previous comment about [query-string](https://www.npmjs.com/package/query-string). It will escape the values in your url to make sure they wont mess up the browser.